server/src/test/java/org/elasticsearch/common/lucene/search/morelikethis/XMoreLikeThisTests.java

```diff
@@ -8,34 +8,136 @@
+                            final char[] buffer = termAtt.buffer();
+                            final int length = termAtt.length();
+                            for (int i = 0; i < length; i++) {
+                                if (buffer[i] == '|') {
+                                    termAtt.setLength(i);
+                                    i++;
+                                    tfAtt.setTermFrequency(ArrayUtil.parseInt(buffer, i, length - i));
+                                    return true;
+                                }
+                            }
+                            return true;
+                        }
+                        return false;
+                    }
+                };
+            }
+        };
+
+        Directory directory = newDirectory();
+        RandomIndexWriter writer = new RandomIndexWriter(random(), directory);
+
+        // Add series of docs with specific information for MoreLikeThis
+        addDoc(writer, "text", "lucene");
+        addDoc(writer, "text", "lucene release");
+        addDoc(writer, "text", "apache");
+        addDoc(writer, "text", "apache lucene");
+
+        // one more time to increase the doc frequencies
+        addDoc(writer, "text", "lucene2");
+        addDoc(writer, "text", "lucene2 release2");
+        addDoc(writer, "text", "apache2");
+        addDoc(writer, "text", "apache2 lucene2");
+
+        addDoc(writer, "text2", "lucene2");
+        addDoc(writer, "text2", "lucene2 release2");
+        addDoc(writer, "text2", "apache2");
+        addDoc(writer, "text2", "apache2 lucene2");
+
+        IndexReader reader = writer.getReader();
+        writer.close();
+        XMoreLikeThis mlt = new XMoreLikeThis(reader, new ClassicSimilarity());
+        mlt.setMinDocFreq(0);
+        mlt.setMinTermFreq(1);
+        mlt.setMinWordLen(1);
+        mlt.setAnalyzer(analyzer);
+        mlt.setFieldNames(new String[] { "text" });
+        mlt.setBoost(true);
+
+        final double boost10 = ((BooleanQuery) mlt.like("text", new StringReader("lucene|10 release|1"))).clauses()
+            .stream()
+            .map(BooleanClause::getQuery)
+            .map(BoostQuery.class::cast)
+            .filter(x -> ((TermQuery) x.getQuery()).getTerm().text().equals("lucene"))
+            .mapToDouble(BoostQuery::getBoost)
+            .sum();
+
+        final double boost1 = ((BooleanQuery) mlt.like("text", new StringReader("lucene|1 release|1"))).clauses()
+            .stream()
+            .map(BooleanClause::getQuery)
+            .map(BoostQuery.class::cast)
+            .filter(x -> ((TermQuery) x.getQuery()).getTerm().text().equals("lucene"))
+            .mapToDouble(BoostQuery::getBoost)
+            .sum();
+
+        // mlt should use the custom frequencies provided by the analyzer so "lucene|10" should be
+        // boosted more than "lucene|1"
+        assertTrue(String.format(Locale.ROOT, "%s should be grater than %s", boost10, boost1), boost10 > boost1);
```

Comment:
```suggestion
        assertTrue(String.format(Locale.ROOT, "%s should be greater than %s", boost10, boost1), boost10 > boost1);
```

I basically just copied from Lucene, so we also inherited spelling mistakes :)